frontend/src/app/modules/intervention-form/file-uploader/file-uploader.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+export enum SupportedFileTypes {
+  Images = 'image/*',
+  Pdf = 'application/pdf',
+  WordDocument = 'application/msword',
+  Text = 'text/plain',
+  Video = 'video/*',
+  Audio = 'audio/*',
+}
+
+@Component({
+  selector: 'eko-file-uploader',
+  templateUrl: './file-uploader.component.html',
+  styleUrls: ['./file-uploader.component.scss'],
+})
+export class FileUploaderComponent implements OnInit {
+  @Input() imagesOnly = false;
+  @Input() editTemporaryDisabled = false;
+
+  files: Array<File> = [];
+  addFilesMessage: string;
+  acceptedFileTypes: string;
+
+  ngOnInit(): void {
+    this.addFilesMessage = `Dodaj ${this.imagesOnly ? 'zdjęcia' : 'pliki'}`;
+    this.acceptedFileTypes = this.imagesOnly ? SupportedFileTypes.Images : this.getAllSupportedTypes();
+  }
+
+  openFileBrowser(input: HTMLInputElement) {
+    input.click();
+  }
+
+  onFilesSelected(input: HTMLInputElement) {
+    const newFiles = this.filterExistingFiles(input.files, this.files);
+    this.files = this.files.concat(newFiles);
+    input.value = '';
```

Comment:
Do we need to clear it? In general manipulating input.value of file input doesn't have any effect due to security restrictions.

Reply:
I believe you are partially right - empty string is the only allowed value for the input.value manipulated from javascript, as far as I remember.
However, with current implementation of the component it does not have any effect on its workings.
I am pretty sure there was a time when I needed it during prototyping phase, but sincerely I do not remember why, as this was more than a month ago :)

I will remove it, good catcht!